src/Foundation/Speedy/code/Settings/SpeedyGenerationSettings.cs

```diff
@@ -86,23 +89,13 @@ public static bool IsCriticalJavascriptEnabledAndPossible(this Item item)
         }
 
         private static Item GetGlobalSettingsItem()
-        {
-            return GetContextDatabase().GetItem(SpeedyConstants.GlobalSettings.SpeedyGlobalSettingsId);
-        }
-
-        private static Item GetGlobalSettingsItemFromMaster()
         {
             return GetMasterDatabase().GetItem(SpeedyConstants.GlobalSettings.SpeedyGlobalSettingsId);
         }
 
-        private static Database GetContextDatabase()
```

Comment:
The Get Context calls are required by the front end on both CD and CM to generate the new layout. Or is there a way around this ?

Reply:
Currently all the Critical Generation is happening on CM. So we can save the item and then publish it? This is used to read the settings for API endpoint and usernames. And when running the Command(Ribbon) it seems to set the Context Database to Core which does not have these settings.

Reply:
A few of the razor files used in the layout lookup a global settings item. Just when its coming from the page being generated on CD it requires the context database. 

Flow: 
Layout -> SpeedyJavascriptLoaderSettings.cshtml -> GetDeferJSLoadForMilliseconds() -> GetGlobalSettingsItem() -> GetContextDatabase()